### PR TITLE
Update IPython notebooks

### PR DESCRIPTION
**Pull request by mliu49**
### Motivation or Problem
This PR updates the notebooks located in `RMG-Py/ipython` to work with Python 3 and RMG API changes.

The only exception is `globalUncertainty.ipynb`, which will be updated along with the MUQ changes in a separate PR.

### Description of Changes
In the process of updating the notebooks, a few bugs were also discovered and fixed.

The parse_source example model also had to be updated to reflect RMG-database changes (in particular, the replacement of the R_Recombination tree with auto-generated groups).

### Testing
I have run all of the notebooks locally. To confirm, the reviewer can do the same. All of the notebooks can be executed in their entirety without changing any user inputs, and should run without any errors.

### CODE REVIEW

**Comment by mliu49** [comment]
@mjohnson541 and @alongd, it would be really nice to get this in before the IPython session on Wednesday. Do you have time to look at it or should I request other reviewers?

**Comment by alongd** [comment]
I'm sorry, I won't have time for a decent review by Wednesday.

File: rmgpy/data/kinetics/family.py [hunk 1]
```diff
@@ -2616,25 +2616,41 @@ def get_labeled_reactants_and_products(self, reactants, products):
         new entities in memory so input molecules `reactants` and `products` won't be affected.
         If RMG cannot find appropriate labels, (None, None) will be returned.
         """
+
```
Comment (amarkpayne) [inline rmgpy/data/kinetics/family.py]: Was this addition intentional?

File: ipython/mechanism_analyzer.ipynb [hunk 2]
```diff
@@ -837,50 +806,57 @@
...
+   "version": "3.7.4"
+  },
+  "pycharm": {
+   "stem_cell": {
+    "cell_type": "raw",
+    "metadata": {
+     "collapsed": false
+    },
+    "source": []
+   }
```
Comment (amarkpayne) [inline ipython/mechanism_analyzer.ipynb]: Are these changes necessary or can they be discarded?

**Comment by amarkpayne** [comment]
Quick comment on running the IPython notebooks: uncommenting the line `job.load_chemkin_model('data/minimal_model/chem_annotated.inp',transport_file='data/minimal_model/tran.dat')` in the 5th code cell of cantera_sensitivity_comparison is broken. It is complaining about a missing file, did it get renamed?

File: ipython/find_parameter_sources_and_assign_uncertainties.ipynb [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Parameter Uncertainty Assignment\n",
+    "\n",
+    "This script aims to extract model sources in a clear and informative format.  The script first shows what all the kinetic and thermo sources are in a model. Then it goes through each reaction and species to show their source and what the assigned uncertainties are.  This can be used with any RMG-generated CHEMKIN file that is annotated."
+   ]
```
Comment (amarkpayne) [inline ipython/find_parameter_sources_and_assign_uncertainties.ipynb]: Should we update this comment to include a warning that this will only work if the model was generated with a version of RMG where the uncertainty comments are compatible with this version of RMG?

File: ipython/combustion_model_and_ignition_delay_demo.ipynb [hunk 4]
```diff
@@ -100,14 +100,14 @@
...
+    "    temperature=({temperature!s},'K'),\n",
+    "    pressure=({pressure!s},'atm'),\n",
+    "    initialMoleFractions={{\n",
+    "        'fuel': {fuel_stoich!s},\n",
     "        'O2': 1,\n",
     "        'N2': 3.76,\n",
-    "    },\n",
-    "    terminationTime=('''+str(sim_time/1000.0)+''','s'),\n",
+    "    }},\n",
+    "    terminationTime=({sim_time/1000.0!s},'s'),\n",
```
[the remark this answers is not in the document]
Comment (mliu49) [reply ipython/combustion_model_and_ignition_delay_demo.ipynb]: For this specific one or in general?

File: ipython/findParameterSourcesAndAssignUncertainies.ipynb [hunk 5]
```diff
@@ -333,27 +326,34 @@
...
-    "uncertainty.kineticInputUncertainties"
+    "uncertainty.kinetic_input_uncertainties"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
```
[the remark this answers is not in the document]
Comment (mliu49) [reply ipython/findParameterSourcesAndAssignUncertainies.ipynb]: I think I added a blank cell at the end. Would you like me to remove it?

File: rmgpy/data/kinetics/family.py (hunk 1 above)
Comment (mliu49) [reply rmgpy/data/kinetics/family.py]: I don't think so. I was moving code around before settling on the final version of the fix.

File: ipython/mechanism_analyzer.ipynb (hunk 2 above)
Comment (mliu49) [reply ipython/mechanism_analyzer.ipynb]: They're probably unnecessary. I opened this on in PyCharm to edit so I could use more advanced functionality. I guess it added additional metadata.

File: ipython/combustion_model_and_ignition_delay_demo.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (amarkpayne) [reply ipython/combustion_model_and_ignition_delay_demo.ipynb]: I think in general for the cases I saw in this file

File: ipython/findParameterSourcesAndAssignUncertainies.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (amarkpayne) [reply ipython/findParameterSourcesAndAssignUncertainies.ipynb]: No that's fine, as long as there is only one blank cell at the end this is preferred.

**Comment by amarkpayne** [comment]
I think this PR is good to go. Rebase to update this and I'll merge it in
